### ISSUE TITLE
Is it possible to check value type inside a template ?

### ISSUE THREAD

**Issue by UnrealKonnord** [hairyhenderson/gomplate#892]
Hello,

I am a new comer inside the golang community.
I want to know if there is a way to know a value type ?
Does golang has something like that ?

```
{{ if is $object dict }}
It's a dict:
  {{ range $key, $value := $object }}
    - "{{ $key }}": {{ $value }}
  {{ end }}
{{ else if is $object number }}
   It's a number: {{ $object }}
{{ end }}
```

It's very useful to write more dynamic template in other language.
The Jinja template has something like that:

```
{% if object is mapping %}
  It's a dict:
  {% for key, value in object.items() %}
    - "{{ key }}": {{ value }}
  {% endfor %}
{% elif object is number %}
It's a number: {{ $object }}
{% endif %}
```

Thank you to reading my question :smile:

**Comment by hairyhenderson** [comment]
Hi @UnrealKonnord, this is a good question!

There is the `printf` function, which you can use to print the type:

```console
$ gomplate -i '{{ printf "%T, %T, %T, %T, %T" 42 true "foo" (slice "a" "b") (dict "foo" "bar") }}'
int, bool, string, []interface {}, map[string]interface {}
```

But it'd probably be more convenient to have some sort of `isMap`/`isString`/`isX` function too. I'll consider this a feature request - should be pretty straightforward to provide. The hardest part will be deciding which namespace the functions will live in 😉 (I'm thinking maybe [test](https://docs.gomplate.ca/functions/test/)? 🤔)

**Comment by UnrealKonnord** [comment]
Thank to your answer it was very helpful.
I think test it's a good place. The function can also take place into the conversion namespace ?
There is already ToInt, ToBool, ToX functions.
It also be confusing because isX is not a conversion but a test.
So I think test is already the best place.

**Comment by hairyhenderson** [comment]
Thanks for the feedback @UnrealKonnord - I'll try to get those in before v3.8 🙂

**Comment by hairyhenderson** [comment]
@UnrealKonnord FYI I've implemented `isKind` in https://github.com/hairyhenderson/gomplate/pull/900 - it'll be available in gomplate 3.8.0 soon
